src/components/inline_edit/inline_edit_form.test.tsx

```diff
@@ -333,5 +333,66 @@ describe('EuiInlineEditForm', () => {
         });
       });
     });
+
+    describe('keyboard events', () => {
+      test('pressing the Enter key saves text and returns to readMode', () => {
+        const { getByTestSubject, getByText } = render(
+          <EuiInlineEditForm
+            {...commonInlineEditFormProps}
+            startWithEditOpen={true}
+          />
+        );
+
+        fireEvent.change(getByTestSubject('euiInlineEditModeInput'), {
+          target: { value: 'New message!' },
+        });
+        fireEvent.keyDown(getByTestSubject('euiInlineEditModeInput'), {
+          key: 'Enter',
+        });
+
+        expect(getByTestSubject('euiInlineReadModeButton')).toBeTruthy();
+        expect(getByText('New message!')).toBeTruthy();
+      });
+
+      test('pressing the Escape key cancels text changes and returns to readMode', () => {
+        const { getByTestSubject, getByText } = render(
+          <EuiInlineEditForm
+            {...commonInlineEditFormProps}
+            startWithEditOpen={true}
+          />
+        );
+
+        fireEvent.change(getByTestSubject('euiInlineEditModeInput'), {
+          target: { value: 'New message!' },
+        });
+        fireEvent.keyDown(getByTestSubject('euiInlineEditModeInput'), {
+          key: 'Escape',
+        });
+
+        expect(getByTestSubject('euiInlineReadModeButton')).toBeTruthy();
+        expect(getByText('Hello World!')).toBeTruthy();
+      });
+
+      it('calls passed `inputModeProps.onKeyDown` callbacks', () => {
+        const onKeyDown = jest.fn();
+
+        const { getByTestSubject } = render(
+          <EuiInlineEditForm
+            {...commonInlineEditFormProps}
+            onSave={onSave}
+            startWithEditOpen={true}
+            editModeProps={{ inputProps: { onKeyDown } }}
+          />
+        );
+
+        fireEvent.keyDown(getByTestSubject('euiInlineEditModeInput'), {
+          key: 'Enter',
+        });
+
+        // If a keyDown event is mapped to Enter/Escape, both the default and custom events should run
+        expect(onKeyDown).toHaveBeenCalled();
+        expect(onSave).toHaveBeenCalled();
```

Comment:
Rather than asserting on `onSave`, let's match the assertions of the previous tests (we're back in read mode and the text has updated).

```suggestion
        // Both EUI and consumer `onKeyDown` events should have run
        expect(onKeyDown).toHaveBeenCalled();
        expect(getByTestSubject('euiInlineReadModeButton')).toBeTruthy();
        expect(getByText('New message!')).toBeTruthy();
      });
```